backends-velox/src/main/java/org/apache/gluten/backend/VeloxBackendJniWrapper.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.gluten.backend;
+
+import org.apache.spark.util.GlutenShutdownManager;
+
+import scala.runtime.BoxedUnit;
+
+public class VeloxBackendJniWrapper {
+
+  public static native void shutdown();
+
+  static {
+    GlutenShutdownManager.addHook(
+        () -> {
+          shutdown();
+          return BoxedUnit.UNIT;
+        });
+  }
+}
```

Comment:
Let's see if we can use existing class `NativeBackendInitializer`. We can rename it if needed. Then see if we call the teardown method through `ListenerApi#on<...>Shutdown`.

Reply:
> Then see if we call the teardown method through ListenerApi#on<...>Shutdown
@zhztheplayer 
UT failed with this implementation https://github.com/apache/incubator-gluten/pull/5657/commits/eaff68e280c9c1b719b30c2939fcb624fe8bc448 Is there any possible reason that you may know? I can reproduce with the `mvn test` command, but cannot reproduce by running the failed suite separately.

Reply:
> > Then see if we call the teardown method through ListenerApi#on<...>Shutdown
> 
> @zhztheplayer UT failed with this implementation [eaff68e](https://github.com/apache/incubator-gluten/commit/eaff68e280c9c1b719b30c2939fcb624fe8bc448) Is there any possible reason that you may know? I can reproduce with the `mvn test` command, but cannot reproduce by running the failed suite separately.

It's probably because `ListenerApi` is called multiple times among different test suite runs. That's a issue should be addressed separately. I'll try reproduce the issue too from my env.